plugins/modules/system/sudoers.py

```diff
@@ -0,0 +1,214 @@
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: sudoers
+short_description: Manage Linux sudoers files
+description:
+    - This module allows for the manipulation of Linux sudoers files
+author:
+    - "Jon Ellis (@JonEllis0) <ellis.jp@gmail.com>"
+options:
+    command:
+        description:
+            - The command allowed by the sudoers rule
+            - Multiple can be added by passing a list of commands
+        type: list
+        elements: str
+    group:
+        required: false
+        description:
+            - Name of the group for the sudoers rule
+            - (cannot be used in conjunction with user)
+        type: str
+    name:
+        required: true
+        description:
+            - Name of the sudoers rule
+        type: str
+    nopassword:
+        required: false
+        description:
+            - Whether a password will be required to run the sudo'd command
+        default: true
+        type: bool
+    state:
+        required: false
+        default: "present"
+        choices: [ present, absent ]
+        description:
+            - Whether the rule should exist or not
+        type: str
+    user:
+        required: false
+        description:
+            - Name of the user for the sudoers rule
+            - (cannot be used in conjunction with group)
+        type: str
+'''
+
+EXAMPLES = '''
+- name: Allow the backup user to sudo /usr/local/bin/backup
+  sudoers:
+    name: allow-backup
+    state: present
+    user: backup
+    command: /usr/local/bin/backup
+
+- name: Allow the monitoring group to run sudo /usr/local/bin/gather-app-metrics without requiring a password
+  sudoers:
+    name: monitor-app
+    group: monitoring
+    command: /usr/local/bin/gather-app-metrics
+
+- name: Allow the alice user to run sudo /bin/systemctl restart my-service or sudo /bin/systemctl reload my-service, but a password is required
+  sudoers:
+    name: alice-service
+    user: alice
+    command:
+      - /bin/systemctl restart my-service
+      - /bin/systemctl reload my-service
+    nopassword: false
+
+- name: Revoke the previous sudo grants given to the alice user
+  sudoers:
+      name: alice-service
+      state: absent
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+import os
+
+
+class Sudoers(object):
+
+    def __init__(self, module):
+        self.module = module
+        self.name = module.params['name']
+        self.user = module.params.get('user')
+        self.group = module.params.get('group')
+        self.state = module.params['state']
+        self.nopassword = bool(module.params.get('nopassword', True))
+        self.file = '/etc/sudoers.d/{filename}'.format(filename=self.name)
+
+        command = module.params['command']
+        if isinstance(command, list):
+            self.commands = command
+        else:
+            self.commands = [command]
```

Comment:
```suggestion
        command = module.params['command'] or []
```
The only case when command isn't a `list` is when it is `None` which would've resulted in `[None]`.

Reply:
... and the `command` parameter already has a default value `[]`, so I believe that: 
```suggestion
        self.commands = module.params['command']
```
is safe for running.

Or, alternatively, a different way to make it work is to remove the `[]` default and add
```
    required_if=[('state', 'present', ['commands'])]
```
to the `argument_specs`, so `commands` would be required when running `present` and it will be `None` (or ignored) when running `absent`.

Reply:
I went with the required_if rule.